Makefile

```diff
@@ -86,137 +127,179 @@ clean_templates:
-	rm -f $(BIN) $(BIN_WINDOWS) $(BIN_LINUX) $(BIN_DARWIN_AMD64) $(BIN_DARWIN_ARM64)
-	rm -f schema/func_yaml-schema.json
-	rm -f coverage.txt
-
-docs:
-	# Generating command reference doc
-	go run docs/generator/main.go
-
-#############
-##@ Templates
-#############
+.PHONY: templates-check
+templates-check: certs ## Check that the generated embedded filesystems are up-to-date
+	@if [[ -n $$(git diff -- schema/func_yaml-schema.json) ]]; \
+		then echo "\nFunction root certs (templates/certs/ca-certificates.crt) are obsolete, please run 'make certs' and commit the result.\n" >&2; \
+	exit 1; fi
 
 # TODO: add linters for other templates
+# NOTE the potential naming confusion with `templates-check` which checks if
+# the generated file is up-to-date, whereas this checks source code via the
+# linter.
 check-templates: check-rust
 
 check-rust: ## Lint Rust templates
 	cd templates/rust/cloudevents && cargo clippy && cargo clean
 	cd templates/rust/http && cargo clippy && cargo clean
 
+.PHONY: test-templates
 test-templates: test-go test-node test-python test-quarkus test-springboot test-rust test-typescript ## Run all template tests
 
+.PHONY: test-go
 test-go: ## Test Go templates
 	cd templates/go/cloudevents && go mod tidy && go test
 	cd templates/go/http && go mod tidy && go test
 
+.PHONY: test-node
 test-node: ## Test Node templates
 	cd templates/node/cloudevents && npm ci && npm test && rm -rf node_modules
 	cd templates/node/http && npm ci && npm test && rm -rf node_modules
 
+.PHONY: test-python
 test-python: ## Test Python templates
 	cd templates/python/cloudevents && pip3 install -r requirements.txt && python3 test_func.py && rm -rf __pycache__
 	cd templates/python/http && python3 test_func.py && rm -rf __pycache__
 
+.PHONY: test-quarkus
 test-quarkus: ## Test Quarkus templates
 	cd templates/quarkus/cloudevents && mvn test && mvn clean
 	cd templates/quarkus/http && mvn test && mvn clean
 
+.PHONY: test-springboot
 test-springboot: ## Test Spring Boot templates
 	cd templates/springboot/cloudevents && mvn test && mvn clean
 	cd templates/springboot/http && mvn test && mvn clean
 
+.PHONY: test-rust
 test-rust: ## Test Rust templates
 	cd templates/rust/cloudevents && cargo test && cargo clean
 	cd templates/rust/http && cargo test && cargo clean
 
+.PHONY: test-typescript
 test-typescript: ## Test Typescript templates
 	cd templates/typescript/cloudevents && npm ci && npm test && rm -rf node_modules build
 	cd templates/typescript/http && npm ci && npm test && rm -rf node_modules build
 
+###############
+##@ Scaffolding
+###############
+
+.PHONY: certs
+certs: templates/certs/ca-certificates.crt ## Ensure the certs exist
+
+templates/certs/ca-certificates.crt:
+	# Updating root certificates
+	curl --output templates/certs/ca-certificates.crt https://curl.se/ca/cacert.pem
+
+.PHONY: certs-check
+certs-check: certs ## Check that root certificates are up-to-date
+	@if [[ -n $$(git diff -- schema/func_yaml-schema.json) ]]; \
```

Comment:
How does this check certs?

Reply:
Renamed this `certs-modified-check` and added better comments 👍🏻

Reply:
I was getting on `schema/func_yaml-schema.json`. Why it is there?

Reply:
I have removed all but the minimal makefile updates and will submit them separately for discussion.